### ISSUE TITLE
Agree to TOS

### ISSUE THREAD

**Issue by Blackhawke** [dannyvankooten/mailchimp-for-wordpress#83]
My legal brain is pestering me that users subscribing through mailchimp should still have to agree to the site term of service and privacy policies. I  have "accept to register" installed for normal user registration, but there doesn't seems to be an equivalent in your plugin. Having that functionality would be nice (the legal beagle would probably say "essential, but you know how they are).

**Comment by dannyvankooten** [comment]
Hey @Blackhawke ,

Thanks for the suggestion! As a matter of fact, this is something we're currently working on. :-)

Should you want a quick way to add an "agree to terms" checkbox right now then you can use something like the following code to get it done.

```html
<p>
	<input type="checkbox" name="agree_to_terms" required /> Agree to our terms & conditions?
</p>
```

By default, this will only perform a client-side check. If you want to check if the checkbox was checked server-side, you can use the following code.

```php
/**
* Performs additional form validation on MailChimp for WP forms
*
* @return true if form is valid, string with error code otherwise
*/
function myprefix_mc4wp_validate_form( $valid ) {
	
	// perform validation here
	if( ! isset( $_POST['agree_to_terms'] ) ) {
		return 'not_agreed_to_terms';
	}

	return true;
}

add_filter( 'mc4wp_valid_form_request', 'myprefix_mc4wp_validate_form' );

/**
* Registers an additional MailChimp for WP error message
*
* @return array
*/
function myprefix_add_mc4wp_error_message( $messages ) {

	$messages['not_agreed_to_terms'] = array(
		'type' => 'error',
		'text' => 'Please agree to our terms and conditions.'
	);

	return $messages;
}

add_filter( 'mc4wp_form_messages', 'myprefix_add_mc4wp_error_message' );
```

We do realise this is a little technical so yes, we're definitely working on making this easier! Thanks again. :)

**Comment by dannyvankooten** [comment]
Closing this in favor of #61! :smiley:

**Comment by Blackhawke** [comment]
Okay, that's cool! I like that as an interim solution. The other thing
that my legal brain trusts requests is the ability to store (and so
prove) that a user/subscriber has agreed to the TOS. So you might want
to put that in the mix for the improvements you're including. :)

On 3/5/15, Danny van Kooten <notifications@github.com> wrote:
> Closed #83.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dannyvankooten/mailchimp-for-wordpress/issues/83#event-247106942


-- 
-=Michael=-
Metaphor Publications
***********************************
Add me to your address book:
http://ourteam.com/mjmatson
***********************************
